src/generate/utils/ApproachHelpers.hpp

```diff
@@ -0,0 +1,28 @@
+//
+// Created by Carolin on 19.06.2020.
+//
+
+#ifndef KICLIENT_APPROACHHELPERS_HPP
+#define KICLIENT_APPROACHHELPERS_HPP
+
+
+#include <datatypes/scenario/Scenario.hpp>
+#include <datatypes/matchconfig/MatchConfig.hpp>
+#include <datatypes/character/CharacterInformation.hpp>
+#include <generate/caro/evalFunctions_caro.hpp>
+
+class ApproachHelpers {
+    public:
+        ApproachHelpers() = default;
+
+        ApproachHelpers(unsigned int difficulty,
+                        const spy::MatchConfig &config,
+                        const spy::scenario::Scenario &scenarioConfig,
+                        const std::vector <spy::character::CharacterInformation> &characterConfig);
+
+        std::optional<evalFunctions_caro> evalFun_caro;
```

Comment:
Macht es nicht mehr Sinn das Interface so zu gestalten, dass man für eine gegebene Difficulty eine evalFun definiert und dann hier nur ein evalFun objekt hat? Und alle evalFunction bräuchen halt das gleiche Interface (item_choice, eq_choice, game_op)

Reply:
Dann würde man aber ein Problem bekommen falls jemand eine Strategie machen will, die nicht nur eine eavlFunction braucht, sondern auch noch irgendwas anderes.
So kann man alles was der Ansatz braucht in den ApproachHelper packen, ohne dass sich die Signatur für den Aufruf in AI.cpp ändert.